drivers/input/i8042prt/registry.c

```diff
@@ -50,6 +50,21 @@ ReadRegistryEntries(
 	DefaultCrashOnCtrlScroll = 0;
 #endif
 
+	OBJECT_ATTRIBUTES objAttrs;
+    InitializeObjectAttributes(&objAttrs,
+                               RegistryPath,
+                               OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+                               NULL,
+                               NULL);
+
+    HANDLE serviceHandle;
+    Status = ZwOpenKey(&serviceHandle, KEY_READ, &objAttrs);
```

Comment:
Do you do that as well for the same reason as in ramdisk?
According to i8042prt this function is called in DriverEntry after the registry has been duplicated and been ensured to be NULL-terminated, so, you could have passed in argument that latter string instead, ensuring that its buffer is indeed NULL-terminated.

Reply:
However, sermouse: https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/input/sermouse/sermouse.c;hb=5099f24373d6f68f4b98241a3ea488595a109dc1#l91
does not do any such duplication. It would perhaps be needed to do such addition there instead.